### PR TITLE
add list datatypes

### PR DESCRIPTION
**Pull request by dotmancando**
add followings data type and associated tests
* IntegerListType
* DoubleListType
* StringListType
* DateTimeListType

### CODE REVIEW

File: schema/data_type.go [hunk 1]
```diff
@@ -15,25 +15,41 @@ const (
 	// IntegerType is our exported const representing integer data types
 	IntegerType = DataType("http://www.w3.org/2001/XMLSchema#integer")
 
+	// IntegerListType is our exported const representing a value that is a comma
+	// separated list of integer instances. Used for transport data values.
+	IntegerListType = DataType("http://purl.org/iot/vocab/thingful#ntegerList")
```
Comment (Popcore) [inline schema/data_type.go]: Very minor: maybe remove the "Used for transport data values." from the comment since it can be used for any Thing type?

**Comment by dotmancando** [comment]
* I've changed the test to subtests. And probe some odd cases. 
* One interesting case is that, in the case of array of string that member contain "," I decide to throw error here because the result list will become unusable

**Comment by dotmancando** [comment]
update re: @smulube suggestion to use CSV package to parse string array instead

File: schema/data_type.go [hunk 2]
```diff
@@ -87,6 +151,35 @@ func Serialize(value interface{}, dataType DataType) (string, error) {
...
+			return "", fmt.Errorf("error writing %v to csv: %v", value, err)
+		}
+
+		w.Flush()
+
+		if err := w.Error(); err != nil {
+			return "", fmt.Errorf("%v", err)
+		}
+		// csv package add newline char at the end of the line, we remove it here
+		return strings.Replace(buf.String(), "\n", "", -1), nil
```
Comment (smulube) [inline schema/data_type.go]: here we can just use:

```
return strings.TrimSpace(buf.String()), nil
```

**Pull request merged**